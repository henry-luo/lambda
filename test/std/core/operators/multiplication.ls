// Test: Multiplication Operations
// Category: core/operators
// Type: positive
// Expected: 50

// Test multiplication with variables
let x = 10
let y = 5
x * y
