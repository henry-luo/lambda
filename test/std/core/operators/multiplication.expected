09:15:47 [NOTICE] Start transpiling test/std/core/operators/multiplication.ls...
[script:list]
  [content:list[3]]
    no declare
    item:
    [let stam:null]
      declare:
      [assign expr:x:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:y:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [binary expr *.5:int]
      [primary expr:int,const:1]
        [ident:x:int,const:1]
      [primary expr:int,const:1]
        [ident:y:int,const:1]
09:15:47 [NOTICE] compiling C code in 'test/std/core/operators/multiplication.ls' to MIR
C2MIR init end           -- 67 usec
test/std/core/operators/multiplication.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2426, parse tokens -- 1458
  C2MIR preprocessor end    -- 538 usec
  C2MIR parser end          -- 710 usec
  C2MIR context checker end -- 792 usec
  C2MIR generator end       -- 847 usec
C2MIR compiler end                -- 880 usec
09:15:47 [NOTICE] Generating native code...
##### Script 'test/std/core/operators/multiplication.ls' executed: #####
50
